README.md

```diff
@@ -396,40 +516,210 @@ Make sure that you have specified `authenticationEndpoint` during SDK initializa
+| setMaxRetries(count: Int)                          | UploadPolicy.Builder | Sets the maximum number of retries for the upload request. Negative value will throw an `IllegalArgumentException`. Defaults to 5.                                                                                               |
+| setRetryBackoff(interval, BackoffPolicy policy)    | UploadPolicy.Builder | Sets the backoff interval in milliseconds and policy (`UploadPolicy.BackoffPolicy.LINEAR` or `UploadPolicy.BackoffPolicy.EXPONENTIAL`) for retry attempts. Defaults to interval of 10000ms and policy of `BackoffPolicy.LINEAR`. |
+
+Example code
+```kotlin
+val policy = UploadPolicy.Builder()
+    .requireNetworkType(UploadPolicy.NetworkType.UNMETERED)
+    .requiresCharging(true)
+    .requiresDeviceIdle(false)
+    .setMaxRetries(5)
+    .setRetryBackoff(60000L, UploadPolicy.BackoffPolicy.EXPONENTIAL)
+    .build()
+```
+
+## Upload preprocessing
+### Image preprocessing
+The `ImagePreprocessor` class encapsulates a set of methods to apply certain transformations to an image before uploading.
+
+`ImagePreprocessor.Builder` class is responsible for building the ImagePreprocess instances. This class provides following methods to access and modify the policy parameters:
+
+| Parameter                                                                                                               | Type                      | Description                                                              |
+|:------------------------------------------------------------------------------------------------------------------------|:--------------------------|:-------------------------------------------------------------------------|
+| limit(width: Int, height: Int)                                                                                          | ImagePreprocessor.Builder | Specifies the maximum width and height of the image                      |
+| crop(p1: Point, p2: Point)                                                                                              | ImagePreprocessor.Builder | Specifies the two points on the diagonal of the rectangle to be cropped. |
+| format(format: [Bitmap.CompressFormat](https://developer.android.com/reference/android/graphics/Bitmap.CompressFormat)) | ImagePreprocessor.Builder | Specify the target image format.                                         |
+| rotate(degrees: Float)                                                                                                  | ImagePreprocessor.Builder | Specify the rotation angle of the target image.                          |
+
+Example code
+```kotlin
+val preprocessor = ImagePreprocessor.Builder()
+    .limit(1280, 720)
+    .format(Bitmap.CompressFormat.WEBP)
+    .rotate(45f)
+    .build()
+```
+### Video preprocessing
+The `VideoPreprocessor` class encapsulates a set of methods to apply certain transformations to an image before uploading.
```

Comment:
Incorrect copy paste and mention of `image` in video section. Also please describe this with more details and a real-world example.

Reply:
Incorrect text has been fixed, and a brief detail has been added.